app/src/main/java/org/wikipedia/savedpages/SavedPageSyncService.java

```diff
@@ -237,33 +246,47 @@ private long savePageFor(@NonNull ReadingListPage page) throws Exception {
                 .flatMap(response -> Observable.just(response.getDisplayTitle()))
                 .onErrorReturnItem(pageTitle.getDisplayText()); // prevent "redirected" or variant issue
 
+        // TODO: remove page lead and maybe load page summary since it needs description for page
         Observable<retrofit2.Response<PageLead>> leadCall = reqPageLead(CacheControl.FORCE_NETWORK, OfflineCacheInterceptor.SAVE_HEADER_SAVE, pageTitle);
-        Observable<retrofit2.Response<PageRemaining>> sectionsCall = reqPageSections(CacheControl.FORCE_NETWORK, OfflineCacheInterceptor.SAVE_HEADER_SAVE, pageTitle);
+        Observable<retrofit2.Response<MediaList>> mediaListCall = reqMediaList(CacheControl.FORCE_NETWORK, OfflineCacheInterceptor.SAVE_HEADER_SAVE, pageTitle);
+        Observable<retrofit2.Response<References>> referencesCall = reqPageReferences(CacheControl.FORCE_NETWORK, OfflineCacheInterceptor.SAVE_HEADER_SAVE, pageTitle);
         final Long[] pageSize = new Long[1];
         final Exception[] exception = new Exception[1];
 
-        Observable.zip(pageSummaryDisplayTextObservable, leadCall, sectionsCall, (summaryDisplayText, leadRsp, sectionsRsp) -> {
+        Observable.zip(pageSummaryDisplayTextObservable, leadCall, mediaListCall, referencesCall, (summaryDisplayText, leadRsp, mediaListRsp, referencesRsp) -> {
             page.title(summaryDisplayText);
             long totalSize = 0;
             totalSize += responseSize(leadRsp);
             page.downloadProgress(LEAD_SECTION_PROGRESS);
             WikipediaApp.getInstance().getBus().post(new PageDownloadEvent(page));
-            page.downloadProgress(SECTIONS_PROGRESS);
+            totalSize += reqMobileHTML(CacheControl.FORCE_NETWORK, OfflineCacheInterceptor.SAVE_HEADER_SAVE, pageTitle);
+            page.downloadProgress(MOBILE_HTML_SECTION_PROGRESS);
+            WikipediaApp.getInstance().getBus().post(new PageDownloadEvent(page));
+            totalSize += responseSize(mediaListRsp);
+            page.downloadProgress(MEDIA_LIST_PROGRESS);
+            WikipediaApp.getInstance().getBus().post(new PageDownloadEvent(page));
+            totalSize += responseSize(referencesRsp);
+            page.downloadProgress(REFERENCES_PROGRESS);
             WikipediaApp.getInstance().getBus().post(new PageDownloadEvent(page));
-            totalSize += responseSize(sectionsRsp);
             Set<String> imageUrls = new HashSet<>(pageImageUrlParser.parse(leadRsp.body()));
```

Comment:
We no longer need to extract image URLs from the Lead response, since we're now getting them from `MediaList`.

Reply:
done!